src/Extensions/ExtensionContributions.re

```diff
@@ -4,130 +4,226 @@
  * Types for VSCode Extension contribution points
  */
 open Oni_Core;
-open Utility;
 open Rench;
 
-module Commands = {
-  [@deriving (show, yojson({strict: false, exn: true}))]
+module Option = Utility.Option;
+module Result = Utility.Result;
+
+module Command = {
+  [@deriving show]
   type t = {
     command: string,
     title: LocalizedToken.t,
-    category: [@default None] option(string),
+    category: option(string),
   };
+
+  let decode =
+    Json.Decode.(
+      field("command", string) >>= command =>
+      field("title", LocalizedToken.decode) >>= title =>
+      field_opt("category", string) >>= category =>
+      succeed({command, title, category})
+    );
```

Comment:
Example of decoder using `field` and monadic bind (`>>=`). Not too bad for small decoders.